code/modules/admin/centcom_ban_db.dm

```diff
@@ -0,0 +1,113 @@
+// CCBDB = CentCom Ban DB
+// This file contains procs for handling all requests to the CCBDB API, hosted by Bobbahbrown
+// API Documentation: https://centcom.melonmesa.com/swagger/index.html
+// Please refer to said documentation before editing any of the stuff in here, otherwise it will likely break
+
+
+/**
+  * CCBDB Lookup Initiator
+  *
+  * Checks the configuration before invoking the request to the CCBDB server.
+  *
+  * Arguments:
+  * * ckey - ckey to be looked up
+  */
+/datum/admins/proc/create_ccbdb_lookup(ckey)
+	if(!config.centcom_ban_db_url)
+		to_chat(usr, "<span class='warning'>The CentCom Ban DB lookup is disabled. Please inform a maintainer or server host.</span>")
+		return
+
+	var/datum/callback/cb = CALLBACK(src, /datum/admins/.proc/ccbdb_lookup_callback, usr, ckey)
+	SShttp.create_async_request(RUSTG_HTTP_METHOD_GET, "[config.centcom_ban_db_url][ckey]", proc_callback=cb)
+
+/**
+  * CCBDB Lookup Callback
+  *
+  * Callback assigned in [/datum/admins/proc/create_ccbdb_lookup] for async operations without a sleep()
+  *
+  * Arguments:
+  * * user - Mob calling the lookup so the UI can be opened
+  * * ckey - Ckey being looked up
+  * * response - [/datum/http_response] passed through from [SShttp]
+  */
+/datum/admins/proc/ccbdb_lookup_callback(mob/user, ckey, datum/http_response/response)
+	// If the admin DC'd during the lookup, dont try and do things
+	if(!user)
+		return
+
+	// Bail if it errored
+	if(response.errored)
+		to_chat(usr, "<span class='warning'>Error connecting to CentCom Ban DB. Please inform a maintainer or server host.</span>")
+		return
+
+	// Bail if the code isnt 200
+	if(response.status_code != 200)
+		to_chat(usr, "<span class='warning'>Error performing CentCom Ban DB lookup (Code: [response.status_code])</span>")
+		return
+
+	var/list/popup_data = list()
+
+	// A body of "[]" means there were no bans on record for the user
+	if(response.body == "\[]")
+		popup_data += "<center><h3>0 bans detected for [ckey]</h3></center>"
+	else
+		var/list/bans = list()
+		// Wrap this in try/catch because JSON is a finnicky bastard
+		try
+			bans = json_decode(response.body)
+			// Yes I threw a ternary in because I am that obsessed over OCD
+			popup_data += "<center><h3>[length(bans)] ban[length(bans) == 1 ? "" : "s"] detected for [ckey]</h3></center>"
+			// These vars are also just for OCD (Adding a <hr> between each ban but the last)
+			var/total_bans = length(bans)
+			var/index = 0
+			for(var/list/ban in bans)
```

Comment:
Can make this typeless. Yet won't matter much

Reply:
Given the infrequent use and minimal performance drop, I am taking readability over the smidge of performance.